src/Components/Server/src/DependencyInjection/ServerRazorComponentsBuilderExtensions.cs

```diff
@@ -158,14 +159,40 @@ internal IEnumerable<RouteEndpointBuilder> GetEndpoints()
 
         }
 
-        private sealed class ServerComponentsSocketFeature(IHttpWebSocketFeature originalFeature) : IHttpWebSocketFeature
+        private sealed class ServerComponentsSocketFeature(
+            IHttpWebSocketFeature originalFeature,
+            HttpContext httpContext,
+            Func<HttpContext, WebSocketAcceptContext, Task>? configureConnection,
+            bool compressionDisabled)
+            : IHttpWebSocketFeature
         {
             public bool IsWebSocketRequest => originalFeature.IsWebSocketRequest;
 
             public Task<WebSocket> AcceptAsync(WebSocketAcceptContext context)
             {
-                context.DangerousEnableCompression = true;
-                return originalFeature.AcceptAsync(context);
+                context.DangerousEnableCompression = !compressionDisabled;
+                if (configureConnection is null)
+                {
+                    return originalFeature.AcceptAsync(context);
+                }
+                else
+                {
+                    var result = configureConnection.Invoke(httpContext, context);
+                    if (result.IsCompleted)
+                    {
+                        return originalFeature.AcceptAsync(context);
+                    }
+                    else
+                    {
+                        return ReturnAwaited(result, context);
+                    }
```

Comment:
The `ReturnAwaited` method would not execute `originalFeature.AcceptAsync` if there was an async exception. Should this be the same for synchronos ones (e.g. `Task.FromException(...)`)?
```suggestion
                    if (!result.IsCompletedSuccessfully)
                    {
                        return ReturnAwaited(result, context);
                    }
                    else
                    {
                        return originalFeature.AcceptAsync(context);
                    }
```

This pattern is also used in several other places e.g. https://github.com/dotnet/aspnetcore/blob/e07adfec59003776702370fe2091e731a323c67f/src/Http/Routing/src/EndpointRoutingMiddleware.cs#L71-L75

Reply:
Yes you are right, this was an oversight on my part.